compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -685,3 +700,29 @@ fn transcribe_metavar_expr<'a>(
     }
     Ok(())
 }
+
+/// Extracts an identifier that can be originated from a `$var:ident` variable or from a token tree.
+fn extract_ident<'a>(
+    cx: &ExtCtxt<'a>,
+    ident: Ident,
+    interp: &FxHashMap<MacroRulesNormalizedIdent, NamedMatch>,
+) -> PResult<'a, String> {
+    if let NamedMatch::MatchedSingle(pnr) = matched_from_ident(cx, ident, interp)? {
+        if let ParseNtResult::Nt(nt) = pnr
+            && let Nonterminal::NtIdent(nt_ident, _) = &nt.0
```

Comment:
No rawness check here.

Reply:
So I wanted you to *add* a rawness check here for consistency with code a couple lines below.
It *may* be reasonable to expect that `concat(r#sta, tic)` will produce `r#static` instead of `static`, so it may make sense to conservatively keep this an error for now.
Or at least add an open question about raw identifiers to the tracking issue.

`concat_idents!` seems to accept raw identifiers, but it never *produces* raw identifiers.